lib/cog/commands/rule/create.ex

```diff
@@ -0,0 +1,68 @@
+defmodule Cog.Commands.Rule.Create do
+  @moduledoc """
+  Create a rule by specifying the entire rule or by using the shorthand for a
+  command and permission.
+
+  USAGE
+    rule create <rule>
+    rule create <command> <permission>
+
+  ARGS
+    rule        Entire rule to create
+    command     Command used to create a rule (used with permission)
+    permission  Permission used to create a rule (used with command)
+
+  EXAMPLES
+    rule create "when command is s3:delete with arg[0] == 'all' must have site:admin"
+    rule create s3:delete site:admin
+  """
+
+  alias Cog.Models.Rule
+  alias Cog.RuleIngestion
+  alias Piper.Permissions.Parser
+
+  def create(%{options: %{"help" => true}}, _args),
+    do: show_usage
+  def create(_req, [command, permission]),
+    do: do_create("when command is '#{command}' must have '#{permission}'")
+  def create(_req, [rule]),
+    do: do_create(rule)
+  def create(_req, _args),
+    do: {:error, {:invalid_args, 1, 2}}
+
+  defp do_create(rule) do
+    with :ok <- validate_permissions(rule),
+         {:ok, rule} <- create_rule(rule),
+         do: {:ok, rule}
+  end
+
+  defp show_usage(error \\ nil),
+    do: {:ok, "usage", %{usage: @moduledoc, error: error}}
+
+  defp validate_permissions(rule) do
```

Comment:
I wonder if rules would be worth putting into a "repository"? Not suggesting that we do that here, but I think we do things like this in other places. It might be worth consolidating.